core/offchain/src/api.rs

```diff
@@ -329,8 +331,13 @@ where
 		}
 	}
 
-	fn sleep_until(&mut self, _deadline: Timestamp) {
-		unavailable_yet::<()>("sleep_until")
+	fn sleep_until(&mut self, deadline: Timestamp) {
+		// Get current timestamp.
+		let now = self.timestamp();
+		// Calculate the diff with the deadline.
+		let diff = deadline.sub(Duration::from_millis(now.unix_millis()));
```

Comment:
There is actually `Timestamp::diff` method that you should use instead converting to millis:
```
let diff = deadline.diff(now);
```
It returns a `Duration` that just needs to be converted to `std::time::Duration`.